### PR TITLE
Adding Token revocation listener

### PR DESCRIPTION
**Pull request by Ashi1993**
## Adding Token revocation listener

> Explain  in a few lines the purpose of this pull request

**Issue link:** *required*

------

### Development Checklist

1. [ ] Built complete solution with pull request in place.
2. [ ] Ran checkstyle plugin with pull request in place.
3. [ ] Ran Findbugs plugin with pull request in place.
4. [ ] Formatted code according to WSO2 code style.

### Secure Development Checklist

1. [ ] Ran FindSecurityBugs plugin and verified report.
2. [ ] Ran Dependency-check plugin and verified report for new dependencies added.
2. [ ] Ran Dependency-check plugin and verified report for dependency version changes.
3. [ ] Have you verify the PR does't commit any keys, passwords, tokens, usernames, or other secrets?
4. [ ] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)?

### Testing Checklist

1. [ ] Written unit tests.
2. [ ] Documented test scenarios(link available in guides).
3. [ ] Written automation tests (link available in guides).

**Automation Test Details**

| Test Suite        | Test Script IDs   |
| ----------------- | ----------------- |
| Integration Suite | *TCXXXXX, TCXXXX* |

### CODE REVIEW

File: components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/BFSIRequestObjectValidationExtension.java [hunk 1]
```diff
@@ -115,10 +115,11 @@ boolean validateIAMConstraints(RequestObject requestObject,
      */
     @Generated(message = "Empty method")
     @Override
-    protected boolean isValidAudience(RequestObject requestObject, OAuth2Parameters oAuth2Parameters) {
+    protected boolean isValidAudience(RequestObject requestObject, OAuth2Parameters oAuth2Parameters)
+            throws RequestObjectException {
 
         // converted to validation layer
-        return true;
+        return super.isValidAudience(requestObject, oAuth2Parameters);
```
Comment (RivinduM) [inline components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/BFSIRequestObjectValidationExtension.java]: We don't need to override this method

Comment (Ashi1993) [reply components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/BFSIRequestObjectValidationExtension.java]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/12/commits/661110fc425c3513e35e90a631382085ca95d3ef

**Review by hasithakn — DISMISSED** [review]

File: components/org.wso2.bfsi.consent.management.service/src/main/java/org/wso2/bfsi/consent/management/service/util/TokenRevocationUtil.java [hunk 2]
```diff
@@ -37,8 +41,8 @@ public static void revokeTokens(DetailedConsentResource detailedConsentResource,
         AuthenticatedUser authenticatedUser = getAuthenticatedUser(userID);
         Set<AccessTokenDO> accessTokenDOSet = getAccessTokenDOSet(detailedConsentResource, authenticatedUser);
 
-        //TODO : Get consent id claim name from configuration
-        String consentIdClaim = "consent_id";
+        String consentIdClaim = ConsentManagementConfigParser.getInstance().getConfiguration()
+                .get(ConsentManagementConstants.CONSENT_ID_CLAIM_NAME).toString();
```
Comment (RivinduM) [inline components/org.wso2.bfsi.consent.management.service/src/main/java/org/wso2/bfsi/consent/management/service/util/TokenRevocationUtil.java]: Isn't there a possibility to NPE?

Comment (Ashi1993) [reply components/org.wso2.bfsi.consent.management.service/src/main/java/org/wso2/bfsi/consent/management/service/util/TokenRevocationUtil.java]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/12/commits/0efffb1b9f4a14998bc2841b8ec470d8aa5ce755

**Review by RivinduM — APPROVED** [review]

**Review by hasithakn — APPROVED** [review]

**Pull request merged**